### DISCUSSION TITLE
Operation Mapping of amino acid residues/nucleotides

### DISCUSSION THREAD

**Discussion by HrishiDhondge** [edamontology/edamontology#844] [Ideas]
I could not find an operation for the mapping of residues from one database to another (the mapping like [SIFTS](https://www.ebi.ac.uk/pdbe/docs/sifts/index.html)). This is especially the case for mapping residues between UniProt and PDB. 
Is there already an operation similar to this one?  If not, is it possible to add this new operation under the Mapping?

**Comment by veitveit** [comment]
Thanks for bringing this up. 

Is it mapping sequences between databases through similarity or using ids to map between entries in the different databases?

From what I see in your example, the connection between PDB and UniProt is given by the sequence AND the position on the sequence. These are kind of automatically mapped by having the sequences matched.

More precisely, what would be the actual computational method in this case ?

**Comment by HrishiDhondge** [comment]
I will try to explain in a short example to be clear and more precise.
UniProt has information about protein sequences and functions. **Insulin receptor** protein can be found in UniProt using **P06213** identifier (accession id). Consider it as a sequence instance of the protein.
This protein can have multiple structures studied by different labs/peoples, and they submit their studied structures in PDB. For the above-mentioned **Insulin receptor** protein, we have around **_forty_** 3-D structures in PDB as of now. All these structures will neither have the same part (sequence) of the protein nor the same numbering as UniProt (not necessarily but in many cases). 
So in such cases, if I wanted to analyse these structures from PDB, the first task is to map the residues in all these structures to get the same part/sequence numbering from each structure.
The PDB ID **4XST** structure contains amino acids from UniProt starting from 28 up to 337. Here, it is not obligatory for PDB structures to follow the exact numbering for amino acids from UniProt. 
The first residue from the structure could be mapped to the 28th residue in UniProt, but the actual numbering of it in the structure could be 30.
So this mapping is to avoid this confusion because of different conventions.


One clear example of this is **2EK1**. The **chain A** from the structure has **Proline (P)** at the **875**th position. The same residue in UniProt id for the corresponding protein is **Q9NTZ6**, and the corresponding residue is at the **855**th position. 
So 855th position in UniProt is the 875th position in the 2EK1 PDB structure. [SIFTS file](https://ftp.ebi.ac.uk/pub/databases/msd/sifts/xml/2ek1.xml.gz)

**Comment by veitveit** [comment]
Your detailed description is superuseful! 

If I understand correctly, this involves a) mapping the sequences between UniProt and PDB to create the links between the IDs (EDAM operation "ID mapping") and then b) linking the residues between the aligned sequences. The latter indeed does not seem to have a corresponding entry in EDAM. 

The question is how relevant this operation will be. If you envision that there will be many (>10) tools using this operation, then a new entry will be great. 

In this case, could you give it a try creating a new issue? The operation template helps providing the details.

**Reply by HrishiDhondge** [reply]
The usage of this operation depends on the developer of the tool. It is quite common in structural bioinformatics to analyze structure-sequence relationships.
